docs/docs/spec.md

```diff
@@ -342,10 +342,11 @@ innermost.
 
 "Enclosing" is defined as follows:
 
--   Class members are enclosed by classes.
--   Classes may be enclosed by other class members or classes.
--   Top-level classes are enclosed by packages.
--   Packages may be enclosed by modules.
+-   Each class member is enclosed by a class.
+-   Each non-top-level class is enclosed by a class or class member.
```

Comment:
What about this:

* Each local or anonymous class is enclosed by the class member in which it is declared.
* Each class member (including member classes) is enclosed by the class in which it is declared.
* Each top-level class is enclosed by its package.
* Each package may be enclosed by a module.
* Modules are not enclosed by anything.

[By the way, how are unexported and unopened packages enclosed by a module? I can't figure it out from the JLS.]

Reply:
Anonymous (and perhaps local) classes aren't necessarily enclosed by a class member:

```
class Foo {
  Object o = new Object() {};
}
```

(And a local class inside a `static {}` block probably isn't enclosed by a "member" in normal JLS terms, though I'd have to check that.)

We might be able to eliminate some overlap between the "class member" and "class" cases by changing "Each class member is enclosed by a class" to "Each method or constructor is enclosed by a class." But it's probably just as well to wait until we can [rework this section more thoroughly](https://github.com/jspecify/jspecify/issues/622).

As for unexported and unopened packages: I would expect that to work like exported packages (though there might be restrictions on what annotation processors can see; I don't know). The JLS probably doesn't need to get into it, since it probably doesn't define "enclosing," so our best reference is probably still [`Element.getEnclosingElement()`](https://docs.oracle.com/en/java/javase/22/docs/api/java.compiler/javax/lang/model/element/Element.html#getEnclosingElement()).